Classes/UVWelcomeViewController.m

```diff
@@ -280,60 +248,44 @@ - (void)logoTapped {
 #pragma mark ===== UISearchBarDelegate Methods =====
 
 - (BOOL)searchBarShouldBeginEditing:(UISearchBar *)searchBar {
-    [searchBar setShowsCancelButton:YES animated:YES];
-    _filter = IA_FILTER_ALL;
-    searchBar.showsScopeBar = YES;
-    searchBar.selectedScopeButtonIndex = 0;
-    [searchBar sizeToFit];
-    
-    if (FORMSHEET) {
-        _tableView.tableHeaderView = searchBar;
-        _searching = YES;
-        [_tableView reloadData];
-    } else {
-        [_searchController setActive:YES animated:YES];
-        _searchController.searchResultsTableView.tableFooterView = [[UIView alloc] initWithFrame:CGRectZero];
-    }
+    _filter = _searchController.searchBar.selectedScopeButtonIndex;
     return YES;
 }
 
 - (void)searchBar:(UISearchBar *)searchBar selectedScopeButtonIndexDidChange:(NSInteger)selectedScope {
     _filter = searchBar.selectedScopeButtonIndex;
-    if (FORMSHEET) {
-        [_tableView reloadData];
-    } else {
-        [_searchController.searchResultsTableView reloadData];
-    }
+    // DDSearch
+    [self updateSearchResultsForSearchController:_searchController];
 }
 
-- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
-    _instantAnswerManager.searchText = searchBar.text;
-    [_instantAnswerManager search];
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
+    // DDSearch
+    _searchController.searchBar.selectedScopeButtonIndex = 0;
+    _searchController.searchBar.text = @"";
+    _instantAnswerManager.instantAnswers = [NSArray array];
+    [_tableView reloadData];
 }
 
-- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar {
-    _searchBar.showsScopeBar = NO;
-    if (FORMSHEET) {
-        [_searchBar setShowsCancelButton:NO animated:YES];
-        _searchBar.text = @"";
-        _instantAnswerManager.instantAnswers = [NSArray array];
-        [_searchBar resignFirstResponder];
-        _searching = NO;
-        [_searchBar sizeToFit];
-        _tableView.tableHeaderView = _searchBar;
-        [_tableView reloadData];
+#pragma mark ==== UISearchResultsUpdating Methods ====
+
+// DDSearch
+- (void)updateSearchResultsForSearchController:(UISearchController *)searchController {
+    _instantAnswerManager.searchText = searchController.searchBar.text;
+    [_instantAnswerManager search];
+    
+    if (_searchController.searchResultsController) {
+        UVWelcomeSearchResultsController *searchResultsTVC = (UVWelcomeSearchResultsController *)_searchController.searchResultsController;
```

Comment:
@austintaylor The reason that I had the call to `updateSearchResultsForSearchController:` in `didUpdateInstantAnswers` is b/c it was never displaying the results since the results were being returned after the search results controller was initialized and attempted to load. 

A fix that I have that seems to be working was to move the code for setting up the search results controller out of `updateSearchResultsForSearchController:` and into `didUpdateInstantAnswers`. The `updateSearchResultsForSearchController:` method still grabs the search text and calls `[_instantAnswerManager search];` and that is it.

`#pragma mark ==== UISearchResultsUpdating Methods ====

- (void)updateSearchResultsForSearchController:(UISearchController *)searchController {
    _instantAnswerManager.searchText = searchController.searchBar.text;
    [_instantAnswerManager search];
}

#pragma mark ===== Search handling =====

- (void)didUpdateInstantAnswers {
    if (_searchController.active && ![_searchController.searchBar.text isEqualToString:@""]) {
        if (_searchController.searchResultsController) {
            UVWelcomeSearchResultsController *searchResultsTVC = (UVWelcomeSearchResultsController *)_searchController.searchResultsController;
            searchResultsTVC.searchResults = self.searchResults;
            searchResultsTVC.tableView.backgroundView = nil;
            searchResultsTVC.tableView.separatorStyle = UITableViewCellSeparatorStyleSingleLine;
            [searchResultsTVC.tableView reloadData];
        }
    }
}`

I then changed the `searchBar:selectedScopeButtonIndexDidChange:` method to call `[self didUpdateInstantAnswers];` instead of `updateSearchResultsForSearchController:` to update the results when the scope is changed.

Hopefully this helps make it a little easier to follow the code. Any other recommendations on this piece?